### ISSUE TITLE
Project using renv that contains .R file with syntax error crashes RStudio when opened

### ISSUE THREAD

**Issue by mb706** [rstudio/rstudio#6946]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.959
    OS Version      : Windows 10
    R Version       : 4.0.0

### Steps to reproduce the problem
1. Create new project (File -> New Project -> New Directory -> New Project -> [enter any name] -> Create Project
2. Run in R console: `renv::init()`
3. Create new file with the content
    ```r
    f <- function() {
      if (TRUE)
    }
    ```
    and save it in the project directory as an `.R` file (e.g. `test.R`)
4. Quit RStudio, not saving workspace image. (Not sure if saving or not saving WS image is important)
5. Open RStudio again; if RStudio doesn't open the projec created in 1. automatically, open it.

### Describe the problem in detail
Trying to open the project gives the following error:
![error](https://user-images.githubusercontent.com/15801081/82734025-50e23780-9d18-11ea-83e9-d21a1e5bd444.png)
followed by another error and a useless RStudio session that says "Cannot Connect to R" and does not react to input.

(for people googling this: )
```
ERROR r error 4 (R code execution error) [errormsg: Error in as.environment(pos) :
  no item called "package:grDevices" on the search list
]; OCCURRED AT class rstudio::core::Error __cdecl
rstudio::r::exec::`anonymous-namespace'::evaluateExpressions
Unsafe(struct SEXPREC *,struct SEXPREC *,struct SEXPREC
**,class rstudio::r::sexp::Protect *,enum
rstudio::r::exec::`anonymous-namespace'::EvalType)
src/cpp/r/RExec.cpp:171
```

I am assuming this is a weird interaction between whatever `renv` does during its initialization and the syntax error in an `.R` file in the project directory.

Editing the `.Rprofile` file and including the line
```r
library("grDevices")
```
before `renv`'s `source("renv/activate.R")` seems to be a workaround.

### Describe the behavior you expected
RStudio should open the project normally.

- [ ] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [X] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Thanks for the bug report and the detailed instructions! I was able to reproduce this -- for what it's worth, only on Windows; not on macOS.

**Comment by kevinushey** [comment]
Looks like this reproduces in general if the `.Rprofile` contains a call to `readline()`, e.g.

```
readline("Enter a response: ")
```

Unfortunately `renv` is inadvertently doing this on initialization -- it detects the syntax error in the document, then warns the user the error exists and asks if the user wants to proceed.

**Comment by R-Hannibal** [comment]
@mb706, thanks again for finding and reporting this issue. 

We have a fix in place in our latest Daily build of the upcoming 1.4 release of RStudio (available [HERE](https://dailies.rstudio.com/)). If you get the chance will you let us know if the issue is resolved for you in that build?

**Comment by kfeinauer** [comment]
Verified fixed 1.4.949 on Windows 10
